pytest_flask/plugin.py

```diff
@@ -26,28 +26,10 @@
 class JSONResponse:
     """Mixin with testing helper methods for JSON responses."""
 
-    @cached_property
-    def json(self):
```

Comment:
Should we deprecate this first before removing? Either way a changelog noticed is in order I think. 😁

Reply:
Ahh it seems this is already implemented in the mixed-in class?

Reply:
Yeah, it seems so. Let's confirm, just for peace of mind.

First let's disable all monkeypatching with a small change in our [patching fixture](https://github.com/pytest-dev/pytest-flask/blob/master/pytest_flask/plugin.py#L81):
```python
@pytest.fixture(autouse=True)
def _monkeypatch_response_class(request, monkeypatch):
 
    return # return here, don't do anything

    if "app" not in request.fixturenames:
        return
    # ...
```
Now, let's see what kind of object we get with a little introspection on [test_json_response.py](https://github.com/pytest-dev/pytest-flask/pull/133/files#diff-ae1ce22fcb54fdef8c463925740ee8ab50e72eb1305724f0d8309afb84e0d0c6R7):
```python
class TestJSONResponse:
    def test_json_response(self, client, accept_json):
        res = client.get(url_for("ping"), headers=accept_json)
        print(type(res))
        # ...
```
so, `res ` is a `<class 'flask.wrappers.Response'>`. If we check the [source code of this class](https://github.com/pallets/flask/blob/master/src/flask/wrappers.py#L90), we can see it's just a wrapper for [werkzeug.wrappers.Response](https://github.com/pallets/werkzeug/blob/master/src/werkzeug/wrappers/response.py#L64)

Let's check its [json](https://github.com/pallets/werkzeug/blob/master/src/werkzeug/wrappers/response.py#L637) property

```python
@property
def json(self) -> t.Optional[t.Any]:
    # ...
    return self.get_json()
```
and the [get_json](https://github.com/pallets/werkzeug/blob/master/src/werkzeug/wrappers/response.py#L645) method
```python
def get_json(self, force: bool = False, silent: bool = False) -> t.Optional[t.Any]:
    if not (force or self.is_json):
        return None

    data = self.get_data()

    try:
        return self.json_module.loads(data)    #  <---- loading json into python object
    except ValueError:
        if not silent:
            raise

        return None
```
In `return self.json_module.loads(data)`, `self.json_module` is actually python's native `json` module. We can see `json_module = json` a [little above](https://github.com/pallets/werkzeug/blob/master/src/werkzeug/wrappers/response.py#L634). So the method is basically applying https://docs.python.org/3/library/json.html#json.loads to the json object.

Finally, let's double check the response object returned by the test client just to be sure
```python
class TestJSONResponse:
    def test_json_response(self, client, accept_json):
        res = client.get(url_for("ping"), headers=accept_json)
        print(inspect.getsource(res.get_json))
        # ...
```
and
```python
def get_json(self, force=False, silent=False, cache=True):
        """Parse :attr:`data` as JSON.
        # ...
```
yes, we get the same method. So, that was pretty much my thought process. It would be nice having someone else checking if it makes sense and that I'm not missing anything :rocket: